### PR TITLE
feat(Icon/react): add component

### PR DESCRIPTION
**Pull request by krishnaacc123**
Add Icon component

### CODE REVIEW

File: src/atoms/Icon/Icon.web.js [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+
+const Icon = ({ size, name, fill, testID, ...rest }) => {
+  const IconComponent = icons[name];
+  const theme = useContext(ThemeContext);
+  return (
+    <IconComponent
+      fill={getColor(theme, fill)}
+      height={iconSize[size]}
+      width={iconSize[size]}
+      test-id={testID}
```
Comment (deorukhkarchaitanya) [inline src/atoms/Icon/Icon.web.js:22]: use ```automation-attributes```

File: src/atoms/Icon/__tests__/Icon.web.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { renderWithTheme } from '../../../_helpers/testing';
+import Icon from '../index';
+
+describe('<Icon />', () => {
+  describe('size', () => {
```
Comment (deorukhkarchaitanya) [inline src/atoms/Icon/__tests__/Icon.web.test.js:6]: Add one for ```fill```?

Comment (krishnaacc123) [reply src/atoms/Icon/__tests__/Icon.web.test.js:6]: added

File: src/atoms/Icon/Icon.web.js [hunk 3]
```diff
@@ -1,3 +1,42 @@
...
+import { getColorKeys, getColor } from '../../_helpers/theme';
+
+const iconSize = {
+  xsmall: 12,
+  small: 16,
+  medium: 20,
+  large: 24,
+};
+
+const Icon = ({ size, name, fill, testID, ...rest }) => {
```
Comment (kamleshchandnani) [inline src/atoms/Icon/Icon.web.js:15]: Shouldn’t this be data-testid? testID is for native

Comment (krishnaacc123) [reply src/atoms/Icon/Icon.web.js:15]: @kamleshchandnani 
the attribute returned from automation method will be `data-testid` but the param name is still testID.

Comment (kamleshchandnani) [reply src/atoms/Icon/Icon.web.js:15]: Sure. so in that case we don't need to expose this prop. This is required for react-native

for web we can just use `{...automation('ds-icon')}` and you can remove this from prop-types and everywhere

File: src/atoms/Icon/Icon.web.js (hunk 1 above)
Comment (krishnaacc123) [reply src/atoms/Icon/Icon.web.js]: added

**Pull request merged**